src/main/java/software/amazon/awssdk/crt/http/HttpConnection.java

```diff
@@ -164,6 +246,15 @@ private void onConnectionShutdown(int errorCode) {
         return shutdownFuture;
     }
 
+    public boolean didConnectSuccessfully() {
+        return connectedFuture.isDone() && !connectedFuture.isCompletedExceptionally();
+    }
+
+
+    public boolean isShutdownComplete() {
+        return shutdownFuture.isDone();
+    }
```

Comment:
We've removed public state-exposing functions from the other languages. These could probably be private.

Reply:
Fixed